### DISCUSSION TITLE
Feature request: add configurable cancel timeout parameter

### DISCUSSION THREAD

**Discussion by t3mi** [community/community#42850] [Actions]
### Select Topic Area

Product Feedback

### Body

**Describe the enhancement**
For our case we need to do some infrastructure cleanup (which was provisioned for the test runs in the same workflow) but with current static 5 min cancelation timeout we unable to do it in time if deployment fails. It would be great if there would be a configurable parameter added, at least for self-hosted runners, which will cover such scenario.

**Code Snippet**
```
...
  runs-on: self-hosted
    timeout-minutes: 60
    cancel-timeout-minutes: 15
    defaults:
...
```
**Current behaviour**
Runner `agent name` did not respond to a cancelation request with 00:05:00.

**References**
[#464](https://github.com/actions/runner/issues/464)

**Comment by bryan-bar** [comment]
@nikola-jokic @mikhailkoliada
Is this officially supported now?

https://github.com/actions/runner/issues/464#issuecomment-2200336766

**Comment by iNoahNothing** [comment]
What are the chances of getting this revived? It is quite annoying to have to wait 5 minutes for a workflow to cancel when a runner goes out of service

**Reply by bryan-bar** [reply]
Still not officially documented but I have been using it so far to expand the timeout and handle the cancellation. I assume you could do the opposite.
